dbclient/dbclient.py

```diff
@@ -146,6 +157,24 @@ def _should_retry_with_new_token(self, raw_results):
         else:
             return False
 
+    def req_session(self):
+        """
+        Creates one new request session per thread with retry adapater
+        """
+        if not hasattr(self._local, "session"):
+            adapter = HTTPAdapter(max_retries=Retry(
+                total=self._retry_total, 
+                backoff_factor=self._retry_backoff,
+                status_forcelist=self.http_retry_codes,
+                allowed_methods=frozenset({'DELETE', 'GET', 'HEAD', 'OPTIONS', 'PATCH', 'POST', 'PUT', 'TRACE' }),
+                raise_on_status=False
```

Comment:
Have we tested where the retry was exhausted? With this boolean set to False, it just returns a response and be logged into error logger and continue on the next calls. 

I wonder if we should just throw instead and stop the task if it fails even with all the number of retries
WDYT?

Reply:
I think we are good here.

I have tested the case where retries are exhausted. In this case it returns the response with the offending http status code and the calling code behaves as if retry were not being used. I like this behavior as it is not disruptive to the current logic.

When I tested with exhausting the retries while importing notebooks, the http status code was propagated and re-thrown by the calling code because 429 is in the list of `http_error_codes`.  From here I could adjust my `--retry-total` command line arg and re-run picking back up at my last notebook checkpoint.

Reply:
sounds good to me